lnwallet/commitment.go

```diff
@@ -365,7 +395,8 @@ func (cb *CommitmentBuilder) createCommitmentTx(ourBalance,
 // spent after a relative block delay or revocation event, and a remote output
 // paying the counterparty within the channel, which can be spent immediately
 // or after a delay depending on the commitment type..
-func CreateCommitTx(fundingOutput wire.TxIn, keyRing *CommitmentKeyRing,
+func CreateCommitTx(chanType channeldb.ChannelType,
```

Comment:
Or can there be a lighter base class above `CommitmentBuilder` that contains just what we need in the funding flow. `CommitmentBuilder` extends that with what else is required for a live channel.